### PR TITLE
Read-me file

### PR DESCRIPTION
**Pull request by miiu96**
- Readme file 
- Implemented a mechanism that will fetch the node and proxy configs from Github
- Added an extra API endpoint that will return the initial wallet keys

### CODE REVIEW

File: .github/workflows/pr-tests.yml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+        uses: actions/setup-go@v2
+        with:
+          go-version: 1.20.7
+        id: go
+
+      - name: Check out code
+        uses: actions/checkout@v2
+      - name: Unit tests
+        run: |
+          go test ./...
```
Comment (iulianpascalau) [inline .github/workflows/pr-tests.yml]: missing empty line

**Review by iulianpascalau — CHANGES_REQUESTED** [review]

File: .github/workflows/pr-tests.yml (hunk 1 above)
Comment (miiu96) [reply .github/workflows/pr-tests.yml]: added

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,35 @@
+FROM golang:1.20.5 as builder
```
[the remark this answers is not in the document]
Comment (miiu96) [reply Dockerfile:1]: changed

File: cmd/chainsimulator/config/config.toml [hunk 3]
```diff
@@ -1,7 +1,16 @@
...
+        # mx-chain-go-repo will be used to fetch the node configs folder
+        mx-chain-go-repo = "https://github.com/multiversx/mx-chain-go"
+        # mx-chain-proxy-go-repo will be used to fetch the proxy configs folder
+        mx-chain-proxy-go-repo = "https://github.com/multiversx/mx-chain-proxy-go"
     [config.logs]
         log-file-life-span-in-mb = 1024 # 1GB
         log-file-life-span-in-sec = 432000 # 5 days
         log-file-prefix = "chain-simulator"
-        logs-path = "logs"
+        logs-path = "logs"
```
[the remark this answers is not in the document]
Comment (miiu96) [reply cmd/chainsimulator/config/config.toml:16]: added

File: README.md [hunk 4]
```diff
@@ -1,2 +1,30 @@
...
+
+## Endpoints
+
+`mx-chain-simulator-go` includes all the [proxy endpoints](https://github.com/multiversx/mx-chain-proxy-go#rest-api-endpoints)
+
+
+Additionally, the simulator offers custom endpoints:
+
+- `simulator/generate-blocks/:num`: This endpoint initiates the generation of a specified number of blocks for each shard.
+- `simulator/initial-wallets`: This endpoint will return the initial wallets (address and private key hex encoded)
```
[the remark this answers is not in the document]
Comment (miiu96) [reply README.md:30]: will extend the README in the next PR.

File: pkg/proxy/configs/configsFetcher.go [hunk 5]
```diff
@@ -0,0 +1,172 @@
...
+
+		return out.Close()
+	})
+}
+
+func folderExists(folderPath string) (bool, error) {
+	_, err := os.Stat(folderPath)
+	if os.IsNotExist(err) {
+		return false, nil
+	} else if err != nil {
```
[the remark this answers is not in the document]
Comment (miiu96) [reply pkg/proxy/configs/configsFetcher.go:167]: removed

File: pkg/proxy/configs/configsFetcher.go [hunk 6]
```diff
@@ -0,0 +1,172 @@
...
+			return err
+		}
+		defer out.Close()
+
+		_, err = io.Copy(out, in)
+		if err != nil {
+			return err
+		}
+
+		return out.Close()
```
[the remark this answers is not in the document]
Comment (miiu96) [reply pkg/proxy/configs/configsFetcher.go:159]: fixed

File: pkg/proxy/configs/configsFetcher_test.go [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+
+	"github.com/stretchr/testify/require"
+)
+
+const (
+	mxNodeRepo  = "https://github.com/multiversx/mx-chain-go"
+	mxProxyRepo = "https://github.com/multiversx/mx-chain-proxy-go"
+)
+
+func TestConfigsFetcher(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (miiu96) [reply pkg/proxy/configs/configsFetcher_test.go:16]: refactored

File: cmd/chainsimulator/main.go [hunk 8]
```diff
@@ -85,37 +88,64 @@ func startChainSimulator(ctx *cli.Context) error {
...
+		return err
+	}
+
+	proxyConfigs := ctx.GlobalString(pathToProxyConfigs.Name)
+	err = configsFetcher.FetchProxyConfigs(buildInfo, proxyConfigs)
+	if err != nil {
+		return err
+	}
+
 	startTime := time.Now().Unix()
```
[the remark this answers is not in the document]
Comment (miiu96) [reply cmd/chainsimulator/main.go:110]: done.

File: pkg/proxy/configs/configs.go [hunk 9]
```diff
@@ -54,8 +59,36 @@ func CreateProxyConfigs(args ArgsProxyConfigs) (*ArgsOutputConfig, error) {
 		})
 	}
 
+	pemFile := path.Join(newConfigsPath, "walletKey.pem")
```
[the remark this answers is not in the document]
Comment (miiu96) [reply pkg/proxy/configs/configs.go:62]: done

File: pkg/proxy/configs/configsFetcher.go [hunk 10]
```diff
@@ -0,0 +1,111 @@
...
+	appProxy = "proxy"
+)
+
+type fetcher struct {
+	gitFetcher      GitHandler
+	mxChainNodeRepo string
+	mxChainProxy    string
+}
+
+func NewConfigsFetcher(mxChainNodeRepo, mxChainProxy string, git GitHandler) (*fetcher, error) {
```
[the remark this answers is not in the document]
Comment (miiu96) [reply pkg/proxy/configs/configsFetcher.go:25]: added

File: testscommon/gitFetcherStub.go [hunk 11]
```diff
@@ -0,0 +1,17 @@
+package testscommon
+
+type GitFetcherStub struct {
```
[the remark this answers is not in the document]
Comment (miiu96) [reply testscommon/gitFetcherStub.go:4]: added

**Review by sstanculeanu — APPROVED** [review]

**Review by iulianpascalau — COMMENTED** [review]
just the extra flags

**Comment by miiu96** [comment]
> just the extra flags

done

**Review by sstanculeanu — APPROVED** [review]

**Review by iulianpascalau — APPROVED** [review]

**Pull request merged**